lib/class.nsync.php

```diff
@@ -43,65 +57,209 @@ public static function add_network_sites() {
+			'source_after' => "Show source of post after displaying content."	
+		];	
+
+		$post_status_setting = [
+			'0' => 'Inherit from the post',
+			'draft' => 'Draft',
+			'publish' => 'Publish',
+			'pending' => 'Pending' 
+		];
+
+		$html .= '<p><label>Select the Post Status
+				  <select name="nsync_options[post_status]">';
+
+		// Let the user check the post
+		$post_status = self::$settings['post_status'];
+
+		foreach ($post_status_setting as $value => $text) {
+			
+			$selected = selected( $post_status, $value, 0 );
+			$html .= '<option value="' . $value . '" ' . $selected  . '>' . $text . '</option>';
+		}
+
+		$html .= '</select></label></p>';			
+			
+		// force check to only users that are also members of this blog. 
+		foreach ($post_settings as $input => $msg) {
+
+			$checked =  checked( ( isset( self::$settings[ $input ] ) && self::$settings[ $input ]), true, 0);
+			$html .=	'<p>
+							<label>
+								<input type="checkbox" name="nsync_options[' . $input . ']" value="1" ' . $checked . ' /> 
+								' . $msg . '
+							</label>
+					    </p>';
+		}		
+		
+		$value  = (isset(self::$settings['source_template'])? self::$settings['source_template'] : Nsync::$default_source_template); 
+		$checked = checked( (isset(self::$settings['source_template']) && self::$settings['source_template'] ), true, 0);
+		$html .= '<p><br>
+					<label> 
+						Template used to display source: 
+					  	<input type="text" name="nsync_options[source_template]" value="' . $value . '" ' . $checked .' class="regular-text" /> 
+				  	</label>
+				  </p>';
+
+		$html .= '<p>
+				  	<em>Valid tags include: {site permalink}, {site name}, {post permalink}, {post date}, {post title}, {post author}</em>
+				 </p>
+				 </div>';
+
+		return self::wrapModule ( "Post Information", $html );
+	}
+	
+   /**
+	* Creates module that shows the list of sites that belong to the current admin.
+	*
+	* @param object  $current_user    the current user's information.
+	* 
+	* @return string     html of the module to be printed.
+	*/
+	public static function createExternalPublishing( $current_user ) {
+
+		$html  = ''; 
+
+		if ( $current_user->ID ) {
 			$user_blogs = get_blogs_of_user( $current_user->ID );
-			if( is_array( $user_blogs ) ): 
-				?>
-				<p style="margin:20px 0 0;">Select the from <em>your current sites</em> that want to enabled the external publishing from.</p>
-				<?php
+
+			if ( is_array( $user_blogs ) ) { 
+
+			   $html  .=  "<p class='list-title'>Select from user " . $current_user->user_nicename . " current sites, that you want to enabled the external publishing from.</p> ";
+				
+				foreach ( $user_blogs as $blog ) {
+					
+					if ( $current_blog_id != $blog->userblog_id ) {
+						$html  .= '<div class="publishing-sites">
+										<label> 
+											<input name="nsync_options[active][]" type="checkbox" value="' . esc_attr( $blog->userblog_id ) . '" /> ' . $blog->blogname . 
```

Comment:
You use `esc_attr()` which is great, but also need to escape the main input text and everything else not escaped here. I won't highlight all of the places where escaping is needed, but every time data is being sent from the server-side to the client-side it needs to be escaped.